### PR TITLE
MXRT600 add runtime idle support

### PR DESCRIPTION
**Pull request by mmahadevan108**
I have tested with `tests/subsys/power/power_mgmt_soc` & `tests/subsys/power/power_mgmt`

### CODE REVIEW

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts [hunk 1]
```diff
@@ -60,6 +60,18 @@
 			label = "User LED_RED";
 		};
 	};
+
+	power-states {
+		idle: idle {
+			compatible = "zephyr,power-state";
+			power-state-name = "runtime-idle";
+			min-residency-us = <1000>;
```
Comment (galak) [inline boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts:68]: Would be good in the commit message to explain how/where the 1000 comes from.

Comment (mmahadevan108) [reply boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts:68]: I am looking for some guidance on this. I picked a value based on what I saw was done for other platforms. I have asked the slack power-management channel as well.

File: soc/arm/nxp_imx/rt6xx/power.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+/* Invoke Low Power/System Off specific Tasks */
+void pm_power_state_set(struct pm_state_info info)
+{
+	/* FIXME: When this function is entered the Kernel has disabled
+	 * interrupts using BASEPRI register. This is incorrect as it prevents
+	 * waking up from any interrupt which priority is not 0. Work around the
+	 * issue and disable interrupts using PRIMASK register as recommended
+	 * by ARM.
+	 */
```
Comment (ceolin) [inline soc/arm/nxp_imx/rt6xx/power.c:21]: Yeah, this is being re-current in arm boards, I need to check what is the reason we are using BASEPRI.

File: soc/arm/nxp_imx/rt6xx/power.c [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+	/* Set PRIMASK */
+	__disable_irq();
+	/* Set BASEPRI to 0 */
+	irq_unlock(0);
+
+	switch (info.state) {
+	case PM_STATE_RUNTIME_IDLE:
+		POWER_EnterSleep();
+		break;
```
Comment (ceolin) [inline soc/arm/nxp_imx/rt6xx/power.c:31]: Are you sure that `POWER_EnterSleep` is correctly mapped to `RUNTIME_IDLE` ? This is a question I haven't see this target documentation. Just looking this API name it seems it matches better with `SUSPEND_IDLE`

**Comment by ceolin** [comment]
The pr structure looks correct, just need clarification on the residency value and the state used.

File: soc/arm/nxp_imx/rt6xx/power.c (hunk 3 above)
Comment (mmahadevan108) [reply soc/arm/nxp_imx/rt6xx/power.c:31]: Below is the definition of Sleep mode for this platform:
```
Sleep mode:
Sleep mode saves a significant amount of power by stopping CM33 CPU execution without affecting peripherals or requiring significant wake-up time. The sleep mode affects the relevant CPU only. The clock to the core is shut off. Peripherals and memories are active and operational.
```

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts (hunk 1 above)
Comment (mmahadevan108) [reply boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts]: Thank you. I have update the `min-residency-us` value and added information in the commit message.

**Comment by mmahadevan108** [comment]
@ceolin Is there any more changes needed for this PR.

**Review by ceolin — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**